notes.md

```diff
@@ -0,0 +1,1900 @@
+# PyTorch 2.0 Release notes
+
+# Highlights
+
+- &lt;Summary to be provided>
+
+# Backwards Incompatible changes
+
+### **Set min supported Python version to 3.8 (#93155)**
+
+Previously the minimum supported version of Python for PyTorch was 3.7. This PR updates the minimum version to require 3.8 in order to install PyTorch. See [Hardware / Software Support ](https://github.com/pytorch/pytorch/blob/893aa5df3f2a475c91ea8eadb1353812e52fb227/RELEASE.md#python) for more information.
+
+### **Drop support for CUDA 10 (#89582)**
+
+This PR updates the minimum CUDA version to 11.0. See the [getting-started](https://pytorch.org/get-started/locally/) for installation or [building from source](https://github.com/pytorch/pytorch#from-source) for more information.
+
+### **Update `torch.tensor` and `nn.Parameter` to serialize all their attributes (#88913)**
+
+Any attribute stored on `torch.tensor` and `torch.nn.Parameter` will be serialized. This aligns the serialization behavior of `torch.nn.Parameter`, `torch.Tensor` and other tensor subclasses
+
+<table>
+<tr>
+<th>1.13</th>
+<th>2.0</th>
+</tr>
+<tr>
+<td>
+
+```Python
+# torch.Tensor behavior
+>>> a = torch.Tensor()
+>>> a.foo = 'hey'
+
+>>> buffer = io.BytesIO()
+>>> torch.save(a, buffer)
+>>> buffer.seek(0)
+>>> b = torch.load(buffer)
+
+>>> print(a.foo)
+hey
+>>> print(b.foo)
+AttributeError: 'Tensor' object has no attribute 'foo'
+
+# torch.nn.Parameter behavior
+>>> a = nn.Parameter()
+>>> a.foo = 'hey'
+
+>>> buffer = io.BytesIO()
+>>> torch.save(a, buffer)
+>>> buffer.seek(0)
+>>> b = torch.load(buffer)
+
+>>> print(a.foo)
+hey
+>>> print(b.foo)
+AttributeError: 'Parameter' object has no attribute 'foo'
+```
+
+</td>
+<td>
+
+```Python
+# torch.Tensor behavior
+a = torch.Tensor()
+a.foo = 'hey'
+
+>>> buffer = io.BytesIO()
+>>> torch.save(a, buffer)
+>>> buffer.seek(0)
+>>> b = torch.load(buffer)
+>>> print(a.foo)
+hey
+>>> print(b.foo)
+hey
+
+# torch.nn.Parameter behavior
+>>> a = nn.Parameter()
+>>> a.foo = 'hey'
+
+>>> buffer = io.BytesIO()
+>>> torch.save(a, buffer)
+>>> buffer.seek(0)
+>>> b = torch.load(buffer)
+>>> print(a.foo)
+hey
+>>> print(b.foo)
+hey
+```
+
+</td>
+</tr>
+</table>
+
+If you have an attribute that you don't want to be serialized you should not store it as an attribute on tensor or Parameter but instead it is recommended to use `torch.utils.weak.WeakTensorKeyDictionary`
+
+```Python
+>>> foo_dict = weak.WeakTensorKeyDictionary()
+>>> foo_dict[a] = 'hey'
+>>> print(foo_dict[a])
+hey
+```
+
+### **Algorithms `{Adadelta, Adagrad, Adam, Adamax, AdamW, ASGD, NAdam, RAdam, RMSProp, RProp, SGD}` default to faster `foreach` implementation when on CUDA + differentiable=`False`**
```

Comment:
why is this BC breaking?  It's not clear.  Are you trying to point out that their might be some small differences in results?  If so, so that explicitly.